File: src/main/java/org/interview/jobtask/shortener/storage/identity/IdentityGenerator.java
package org.interview.jobtask.shortener.storage.identity;

import java.util.concurrent.ThreadLocalRandom;

public class IdentityGenerator {

    public final String alphabet;
    private static final int KEY_SIZE = 5;

    private ThreadLocalRandom random = ThreadLocalRandom.current();

    private IdentityGenerator(String alphabet) {
        this.alphabet = alphabet;
    }

    public static IdentityGenerator of(String alphabet) {
        return new IdentityGenerator(alphabet);
    }

    public String generate() {
        String key = "";
        for (int i = 0; i < KEY_SIZE; i++) {
            key += alphabet.charAt(random.nextInt(alphabet.length()));
        }
        return key;
    }

}

File: src/main/kotlin/org/interview/jobtask/shortener/storage/identity/kt/IdentityGenerator.kt
package org.interview.jobtask.shortener.org.interview.jobtask.shortener.storage.identity.kt

import java.util.concurrent.ThreadLocalRandom

class IdentityGenerator private constructor(private val alphabet: String) {

    private val random = ThreadLocalRandom.current()

    companion object {
        fun of(alphabet: String) = IdentityGenerator(alphabet)
        private const val KEY_SIZE = 5
    }

    fun generate() = buildString {
        for (i in 1..KEY_SIZE) {
            append(alphabet[random.nextInt(alphabet.length)])
        }
    }
}